src/components/transport_manager/src/transport_adapter/transport_adapter_impl.cc

```diff
@@ -281,9 +281,11 @@ TransportAdapter::Error TransportAdapterImpl::DisconnectDevice(
        j != to_disconnect.end();
        ++j) {
     ConnectionInfo info = *j;
-    if (OK != info.connection->Disconnect()) {
-      error = FAIL;
-      LOG4CXX_ERROR(logger_, "Error on disconnect " << error);
+    if (info.connection) {
```

Comment:
@dcherniev What if here info.connection is empty and in next moment in different thread it will be fulfilled?